test/WebJobs.Script.Tests/Configuration/AggregatorConfigurationTests.cs

```diff
@@ -16,17 +20,42 @@ namespace Microsoft.Azure.WebJobs.Script.Tests.Configuration
 {
     public class AggregatorConfigurationTests
     {
+        private static readonly string AggregatorPath = ConfigurationPath.Combine(ConfigurationSectionNames.JobHost, ConfigurationSectionNames.Aggregator);
+
+        [Fact]
+        public void Aggregator_Registered_ByDefault()
+        {
+            IHost host = new HostBuilder()
+                .ConfigureDefaultTestWebScriptHost()
+                .Build();
+
+            // Make sure there are two providers registered, and that one has a type with
+            // "FunctionResultAggregatorProvider" (since it is internal to WebJobs)
+            var eventCollectorProviders = host.Services.GetServices<IEventCollectorProvider>();
+            Assert.Equal(2, eventCollectorProviders.Count());
+            Assert.Single(eventCollectorProviders.OfType<FunctionInstanceCollectorProvider>());
+            Assert.Single(eventCollectorProviders.Where(p => p.GetType().Name.Contains("FunctionResultAggregatorProvider")));
+
+            // Also make sure that when requesting the collectors, we end up with a composite that
+            // includes both of the collectors above. This prevents any overriding of the IEventCollectorFactory
+            var eventLogger = host.Services.GetServices<IAsyncCollector<FunctionInstanceLogEntry>>().Single();
+            var field = eventLogger.GetType().GetField("_collectors", BindingFlags.NonPublic | BindingFlags.Instance);
+            var collectors = (IEnumerable<IAsyncCollector<FunctionInstanceLogEntry>>)field.GetValue(eventLogger);
+            Assert.Equal(2, collectors.Count());
+            Assert.Single(collectors.OfType<FunctionInstanceLogger>());
+            Assert.Single(collectors.Where(p => p.GetType().Name.Contains("FunctionResultAggregator")));
+        }
```

Comment:
I don't really like testing with private reflection, but this makes sure that we don't overwrite the `IEventCollectorProvider`. Another test makes sure that FucntionAggregator is working, but this will alert us to exactly what happened.

Reply:
I think instances like this are fine when you have to do it :)